nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/AbstractAzureCosmosDBProcessor.java

```diff
@@ -0,0 +1,337 @@
+    .description("All FlowFiles that cannot be written to Azure Cosmos DB are routed to this relationship").build();
+
+    static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("All input FlowFiles that are part of a successful query execution go here.")
+            .build();
+
+    static final PropertyDescriptor CONNECTION_SERVICE = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-db-connection-service")
+        .displayName("Azure Cosmos DB Connection Service")
+        .description("If configured, this property will use the assigned for retrieving connection string info.")
+        .required(false)
+        .identifiesControllerService(AzureCosmosDBConnectionService.class)
+        .build();
+
+
+    static final PropertyDescriptor URI = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-db-uri")
+        .displayName("Azure Cosmos DB URI")
+        .description("Azure Cosmos DB URI, typically of the form: https://{databaseaccount}.documents.azure.com:443/. Note this host URL is for Azure Cosmos DB with CORE SQL API")
+        .required(false)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor DB_ACCESS_KEY = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-db-key")
+        .displayName("Azure Cosmos DB Access Key")
+        .description("Azure Cosmos DB Access Key from Azure Portal (Settings->Keys)")
+        .required(false)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .sensitive(true)
+        .build();
+
+    static final PropertyDescriptor DATABASE_NAME = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-db-name")
+        .displayName("Azure Cosmos DB Name")
+        .description("The name of the database to use")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor CONTAINER_ID = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-container-id")
+        .displayName("Azure Cosmos Container ID")
+        .description("Azure Cosmos DB Container Id to use")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor CONSISTENCY = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-consistency-level")
+        .displayName("Azure Cosmos DB Consistency Level")
+        .description("Azure Cosmos DB Consistency Level to use")
+        .required(false)
+        .defaultValue(CONSISTENCY_SESSION)
+        .allowableValues(CONSISTENCY_STRONG, CONSISTENCY_BOUNDED_STALENESS, CONSISTENCY_SESSION,
+                CONSISTENCY_CONSISTENT_PREFIX, CONSISTENCY_EVENTUAL)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor PARTITION_KEY = new PropertyDescriptor.Builder()
```

Comment:
Similar comment, how does this impact usage?

Reply:
done